chat/rich-links/src/RichLink/RichLink.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { forwardRef } from 'react';
+
+import Card from '@leafygreen-ui/card';
+import { cx } from '@leafygreen-ui/emotion';
+import { useDarkMode } from '@leafygreen-ui/leafygreen-provider';
+import { Body } from '@leafygreen-ui/typography';
+
+import {
+  badgeAreaStyles,
+  baseStyles,
+  imageBackgroundStyles,
+  richLinkTextClassName,
+  themeStyles,
+} from './RichLink.styles';
+import {
+  RichLinkBadgeControlProps,
+  type RichLinkProps,
+} from './RichLink.types';
+import { RichLinkBadge } from './RichLinkBadge';
+import { richLinkVariants } from './RichLinkVariants';
+
+export const RichLink = forwardRef<HTMLAnchorElement, RichLinkProps>(
+  ({ darkMode: darkModeProp, ...props }, ref) => {
+    const { darkMode, theme } = useDarkMode(darkModeProp);
+
+    if ('variant' in props) {
+      const { variant, ...variantProps } = props;
+      const RichLinkVariant = richLinkVariants[variant];
+      return <RichLinkVariant darkMode={darkMode} {...variantProps} />;
+    }
+
+    const badgeDefaults: Partial<RichLinkBadgeControlProps> = {
+      badgeGlyph: undefined,
+      badgeLabel: undefined,
+      badgeColor: undefined,
+    };
+
+    const {
+      children,
+      imageUrl,
+      badgeGlyph,
+      badgeLabel,
+      badgeColor,
+      href,
+      anchorProps,
+    } = {
+      ...badgeDefaults,
+      ...props,
+    };
+
+    const showBadge = badgeLabel !== undefined;
+
+    const showImageBackground = (imageUrl?.length ?? -1) > 0;
+
+    return (
+      // @ts-expect-error - Card does not correctly infer props based on `as` prop
+      <Card
+        darkMode={darkMode}
+        ref={ref}
+        className={cx(baseStyles, themeStyles[theme], {
+          [badgeAreaStyles]: showBadge,
+          [imageBackgroundStyles(imageUrl ?? '')]: showImageBackground,
+        })}
+        as="a"
+        href={href}
+        target="_blank"
+        {...anchorProps}
```

Comment:
this is not an ideal solution but what if we assert that anchorProps are typed as a div just to get rid of the TS error 
<img width="556" alt="Screenshot 2024-05-09 at 3 38 47 PM" src="https://github.com/mongodb/leafygreen-ui/assets/5740159/72ed5f0b-e6fc-46df-940d-be63a3043caa">

Reply:
Good idea looks like this cast works as expected!